sigle/src/modules/layout/components/HeaderDropdown.tsx

```diff
@@ -0,0 +1,249 @@
+              }
+              css={{
+                width: 'auto',
+                height: '100%',
+                maxWidth: 24,
+                maxHeight: 24,
+                objectFit: 'cover',
+              }}
+            />
+          </ImageContainer>
+          <Typography size="subheading">
+            {settings?.siteName ? settings.siteName : user?.username}
+          </Typography>
+          <StyledChevron css={{ color: '$gray11' }} />
+        </Button>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent sideOffset={8}>
+        <Link href={`/${user?.username}`} passHref>
+          <DropdownMenuItem
+            css={{ py: '$1', px: '$2' }}
+            selected={router.pathname === user?.username}
+            as="a"
+          >
+            <ImageContainer
+              css={{
+                width: 38,
+                height: 38,
+              }}
+            >
+              <Box
+                as="img"
+                src={
+                  settings?.siteLogo
+                    ? settings.siteLogo
+                    : generateAvatar(userAddress)
+                }
+                css={{
+                  width: 'auto',
+                  height: '100%',
+                  maxWidth: 38,
+                  maxHeight: 38,
+                  objectFit: 'cover',
+                }}
+              />
+            </ImageContainer>
+            <Flex direction="column" align="start" justify="center">
+              {settings?.siteName && (
+                <Typography size="subheading" css={{ color: '$gray11' }}>
+                  {settings.siteName}
+                </Typography>
+              )}
+              <Typography size="subparagraph" css={{ color: '$gray9' }}>
+                {user?.username}
+              </Typography>
+            </Flex>
+          </DropdownMenuItem>
+        </Link>
+        <DropdownMenuSeparator />
+        <DropdownMenuItem
+          css={{ color: '$gray11' }}
+          disabled={loadingCreate}
+          onClick={handleCreateNewPrivateStory}
+          onSelect={(e) => e.preventDefault()}
+        >
+          <FileTextIcon />
+          {!loadingCreate ? `Write a story` : `Creating new story...`}
+        </DropdownMenuItem>
+        {upperNavItems.map((item) => (
+          <Link key={item.path} href={item.path} passHref>
+            <DropdownMenuItem
+              css={{ color: '$gray11' }}
+              selected={router.pathname === item.path}
+              as="a"
+            >
+              {item.icon}
+              {item.name}
+            </DropdownMenuItem>
+          </Link>
+        ))}
+        <DropdownMenuSeparator />
+        {lowerNavItems.map((item) => (
+          <Link key={item.path} href={item.path} passHref>
+            <DropdownMenuItem selected={router.pathname === item.path} as="a">
+              {item.name}
+            </DropdownMenuItem>
+          </Link>
+        ))}
+        {!userSubscription && (
```

Comment:
now that we're doing the check here, is it worth just inlining the above element as it's now the only one in the lowerNavItems array?

Reply:
Maybe smth that can be done is to push the upgrade object to the lowerNavItems item if there is no subscription, that way you don't duplicate the code and keep lowerNavItems. Does it make sense?

Reply:
okay yeah that makes sense!